src/rpc.rs

```diff
@@ -171,211 +169,119 @@ impl RpcService {
-        request: AppCallRequest,
-    ) -> Result<AppCallResponse, anyhow::Error> {
-        match request.command.as_str() {
-            "ping" => Ok(AppCallResponse {
-                success: true,
-                message: "pong".to_string(),
-            }),
-            "echo" => Ok(AppCallResponse {
-                success: true,
-                message: request.payload,
-            }),
-            _ => Ok(AppCallResponse {
-                success: false,
-                message: format!("Unknown command: {}", request.command),
-            }),
-        }
-    }
-}
-
-pub async fn start_rpc_server(backend: Backend, addr: &str) -> Result<()> {
-    // Load known groups from the backend
-    backend.load_known_groups().await?;
-    // Write keys to file
-    // Question: which keys should be loaded here?
-    //let base_dir = std::env::current_dir().expect("Failed to get current directory");
-    //let keys_path = base_dir.join("group_keys.txt");
-    //fs::write(&keys_path, keys).expect("Failed to write group keys to disk");
-
-    let rpc_service = RpcService::from_backend(&backend).await?;
-
-    // Print the DHT key and secret
-    let dht_record = rpc_service.descriptor.get_dht_record();
-    let dht_key = dht_record.key();
-    println!("DHT Key: {:?}", dht_key.value);
-    
-    if let Some(secret) = rpc_service.descriptor.get_secret_key() {
-        println!("Secret Key: {:?}", secret);
-    } else {
-        println!("Secret Key: None");
-    }
-
-
-    // Start the update listener
-    rpc_service.start_update_listener().await?;
-
-    // Build the reflection service
-    let reflection_service = Builder::configure()
-        .register_encoded_file_descriptor_set(FILE_DESCRIPTOR_SET)
-        .build_v1()?;
-
-    Server::builder()
-        .add_service(rpc_server::RpcServer::new(rpc_service))
-        .add_service(reflection_service)
-        .serve(addr.parse()?)
-        .await?;
-    Ok(())
-}
-
-#[tonic::async_trait]
-impl rpc_server::Rpc for RpcService {
     async fn replicate_group(
         &self,
-        request: Request<ReplicateGroupRequest>,
-    ) -> Result<Response<ReplicateGroupResponse>, Status> {
-        let group_id = request.into_inner().group_id;
+        request: ReplicateGroupRequest,
+    ) -> Result<ReplicateGroupResponse> {
+        let group_id = request.group_id;
         info!("Replicating group with ID: {}", group_id);
 
-        // Decode the Base64URL-encoded group ID into bytes
-        let group_bytes = URL_SAFE_NO_PAD.decode(&group_id).map_err(|_| {
-            error!("Invalid group ID encoding");
-            Status::invalid_argument("Invalid group ID encoding")
-        })?;
-
-        // Convert group_bytes to [u8; 32]
-        let group_bytes: [u8; 32] = group_bytes.try_into().map_err(|_| {
-            error!("Invalid group key length");
-            Status::invalid_argument("Invalid group key length")
-        })?;
-
+        let group_bytes = URL_SAFE_NO_PAD.decode(&group_id)?;
```

Comment:
Instead of replicate_group lets call it joing_group and take a URL instead of an ID, then call `backend.join_group_from_url()` and replicate the returned group.

We should also replicate all groups on boot by loading them from disk in the backend then iterating the list

Reply:
Renamed to `join_group`
It calls `backend.join_from_url` 
f2073d43319e77398bce39621e4f56108c155370

Reply:
I also created `replicate_known_groups` ca1f0f6601b46554a23580502679e5e205963cec
and call it in the main function c4d26132e507326f13a1bef5a357afce7e91c9b9